### PR TITLE
PTX-2758 Autopilot validation and basic regression test

### PR DESCRIPTION
**Pull request by nikolaypopov**
Stork basic regression test and various improvements for validation of Stork components and images

* Added Autopilot basic regression test that covers multiple Autopilot test cases
             Autopilot - Validate Autopilot is not enabled by default
             Autopilot - Deploy Autopilot with enabled set to true
             Autopilot - Validate Autopilot serviceaccount, clusterrole, clusterrolebindings and configmap when Autopilot is Enabled 
             Autopilot - Validate Autopilot serviceaccount, clusterrole, clusterrolebindings and configmap are deleted when Autopilot is Disabled or is nil
             Autopilot - Delete Autopilot pod and validate it get re-deployed 
             Autopilot - Disable Autopilot by setting enabled to false 
             Autopilot - Delete Autopilot all together by removing autopilot block/object or setting it to nil 
             Autopilot - Autopilot should not be deployed if enabled set to false 
* Added more in-depth validation for Autopilot components and images

Signed-off-by: nikolaypopov <nikolay.popov86@gmail.com>

### CODE REVIEW

File: pkg/util/test/util.go [hunk 1]
```diff
@@ -937,6 +918,83 @@ func ValidateStork(pxImageList map[string]string, cluster *corev1.StorageCluster
...
+		}
+
+		// Validate Autopilot ClusterRoleBinding
+		_, err = rbacops.Instance().GetClusterRoleBinding(autopilotDp.Name)
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("failed to validate ClusterRoleBinding %s, Err: %v", autopilotDp.Name, err)
+		}
+
+		// Validate Autopilot ConfigMap
+		_, err = coreops.Instance().GetConfigMap(autopilotConfigMapName, autopilotDp.Namespace)
```
Comment (piyush-nimbalkar) [inline pkg/util/test/util.go:965]: Also, validate the `autopilot` service account both here and in the else block.

File: test/integration_test/basic_test.go [hunk 2]
```diff
@@ -410,3 +418,64 @@ func BasicStorkRegression(tc *types.TestCase) func(*testing.T) {
 		ci_utils.UninstallAndValidateStorageCluster(cluster, t)
 	}
 }
+
+// BasicAutopilotRegression test includes the following steps:
+// 1. Deploy PX and validate Stork components and images
```
Comment (piyush-nimbalkar) [inline test/integration_test/basic_test.go:423]: *Autopilot

File: test/integration_test/basic_test.go [hunk 3]
```diff
@@ -410,3 +418,64 @@ func BasicStorkRegression(tc *types.TestCase) func(*testing.T) {
...
+// 2. Validate Autopilot is disabled by default
+// 3. Enabled Autopilot and validate Autopilot components and images
```
Comment (piyush-nimbalkar) [inline test/integration_test/basic_test.go:425]: *Enable

File: test/integration_test/basic_test.go [hunk 4]
```diff
@@ -410,3 +418,64 @@ func BasicStorkRegression(tc *types.TestCase) func(*testing.T) {
...
+
+		// Validate Autopilot block is nil
+		require.Nil(t, cluster.Spec.Autopilot, "failed to validate Autopilot block, it should be nil by default, but it seems there is something set in there %+v", cluster.Spec.Autopilot)
+
+		logrus.Info("Enable Autopilot and validate StorageCluster")
+		updateParamFunc := func(cluster *corev1.StorageCluster) *corev1.StorageCluster {
+			// At this point this object should be <nil>
+			if cluster.Spec.Autopilot == nil {
+				cluster.Spec.Autopilot = &corev1.AutopilotSpec{}
+			}
```
Comment (piyush-nimbalkar) [inline test/integration_test/basic_test.go:447]: nit: As we know the object is going to be nil, we can simply set the Enabled flag - 
```
cluster.Spec.Autopilot = &corev1.AutopilotSpec{
  Enabled: true
}
```

**Review by piyush-nimbalkar — COMMENTED** [review]
Looks good. 
Some minor comments.

File: pkg/util/test/util.go (hunk 1 above)
Comment (nikolaypopov) [reply pkg/util/test/util.go:965]: Added check for that

File: test/integration_test/basic_test.go (hunk 3 above)
Comment (nikolaypopov) [reply test/integration_test/basic_test.go]: Fixed

File: test/integration_test/basic_test.go (hunk 2 above)
Comment (nikolaypopov) [reply test/integration_test/basic_test.go]: Fixed

File: test/integration_test/basic_test.go (hunk 4 above)
Comment (nikolaypopov) [reply test/integration_test/basic_test.go]: Changed this to suggested

**Review by piyush-nimbalkar — APPROVED** [review]

**Pull request merged**